### DISCUSSION TITLE
Sticky headers on tables

### DISCUSSION THREAD

**Discussion by PoojMenon** [twilio-labs/paste#1517] [Q&A] [answered]
Hello, are there any plans to enable sticky headers on Tables? And if not, any guidance how we could go about achieving this if needed?

**Comment by zahnster** [comment]
Hi @PoojMenon,

We likely can accomplish this with using position: sticky, but we'll need a code change to our Table component to support it.

I'll finalize a plan of action with the team and get back to you early next week with a timeline. I expect it will only be a small change on our end.

**Comment by richbachman** [comment]
Hi @PoojMenon. We're going to look into adding a `sticky` prop to the table headers. I've added a ticket to our backlog. We'll update you as soon as we know more.

**Reply by PoojMenon** [reply]
Thanks very much for the update @richbachman!

**Comment by sivapriya-k** [comment]
We have a use case where this is required too. @richbachman any rough estimate on when this will be picked up? Thanks.

**Reply by gloriliale** [reply]
Hi @sivapriya-k! This isn't in our next sprint, but we'll have a better estimate after we do some refinement post-sprint. Can you tell me more about the use case you're thinking of?

**Comment by PoojMenon** [comment]
Hello! Just wanted to follow up to see if you have an update on when this will be available?

**Reply by nkrantz** [reply]
Hi @PoojMenon, sorry about the delay on this feature. It seems to have gotten buried a bit in our backlog. I just brought it up with the team and we're going to prioritize it after our current release (10/5). Here's the ticket so that you can follow along: https://issues.corp.twilio.com/browse/DSYS-2842. I've also added a label to this discussion so that it won't get lost again. Thanks!

**Comment by PoojMenon** [comment]
We'll be using the new Data Grid component instead of a Table here - any chance this will also be done for the Data Grid?

**Reply by SiTaggart** [reply]
Still on the roadmap, but it will likely be available for both Table and Data Grid, given Data Grid inherits and shares a lot from Table.

**Comment by andoyle** [comment]
Hey folks, any update on this? I can't see it on the Roadmap: https://paste.twilio.design/roadmap

We've got users requesting this now:

> At times we provide screenshots of breakdowns to customers and share our screen on calls with customers for education purposes. This will allow for information to be clear and understandable.

**Reply by gloriliale** [reply]
Hi @andoyle! I've surfaced this to the team and will get back to you as soon as I've got more insight from them. Thank you!

**Reply by andoyle** [reply]
Thanks!

**Reply by gloriliale** [reply]
hi @andoyle! So unfortunately it looks like resource constraints mean we won't be getting to sticky headers until Q3 or Q4 of this year :( . However, if your team needs it sooner than that and is willing to contribute that feature, then that might be something we can discuss. Let me know your thoughts!

**Reply by andoyle** [reply]
Okay, no worries. I'll run it by the team and let you know 👍

**Comment by javiruiz01** [comment]
Hi team! I'm sorry to ask the same question one more time, but are there any updates on this? I checked the ticket and it is still open, so wondering if you had talked about this recently ^^ Thanks!

**Reply by kate-porter** [reply]
Hey @javiruiz01 - thanks for checking in here. We're actually working on this in our current sprint! [Here's the ticket](https://issues.corp.twilio.com/browse/DSYS-2842) if you'd like to follow along. We'll follow up on this discussion once it's complete.

**Reply by javiruiz01** [reply]
Awesome! Let us know if you need any help from our side 👍

**Reply by TheSisb** [reply]
There's a new PR up to add this feature https://github.com/twilio-labs/paste/pull/2574. I expect it to land in our component sometime in the next week.

**Comment by SiTaggart** [comment, marked as answer]
Hey @javiruiz01, @andoyle and friends

Sticky table and data grid headers just landed in Paste. They are ready to install from your nearest NPM registry!

Thanks for hanging in there
